render-test/render_test.cpp

```diff
@@ -40,7 +40,30 @@ void operator delete(void* ptr, size_t) noexcept {
 #endif
 
 namespace {
-using ArgumentsTuple = std::tuple<bool, bool, uint32_t, std::string, std::vector<std::string>, std::string>;
+
+template <typename T>
+TestRunner::UpdateResults initUpdateResults(T& testUpdateResultsValue) {
+    if (!testUpdateResultsValue) {
+#if !TEST_READ_ONLY
+        if (getenv("UPDATE_DEFAULT")) return TestRunner::UpdateResults::DEFAULT;
+        if (getenv("UPDATE_PLATFORM")) return TestRunner::UpdateResults::PLATFORM;
+        if (getenv("UPDATE_METRICS")) return TestRunner::UpdateResults::METRICS;
+#endif
+        return TestRunner::UpdateResults::NO;
+    }
+    std::string str = args::get(testUpdateResultsValue);
+#if !TEST_READ_ONLY
+    if (str == "default") return TestRunner::UpdateResults::DEFAULT;
+    if (str == "platform") return TestRunner::UpdateResults::PLATFORM;
+    if (str == "metrics") return TestRunner::UpdateResults::METRICS;
+#endif
+    mbgl::Log::Warning(
+        mbgl::Event::General, "Unsupported update test results mode: \"%s\" will be ignored", str.c_str());
```

Comment:
I think this line should be moved into #if #endif scope.

Reply:
The intention here is to emit warning in any case the given mode cannot be supported, including the case when test runner is build with `TEST_READ_ONLY` flag